### DISCUSSION TITLE
Links to source code

### DISCUSSION THREAD

**Discussion by wench** [exult/exult#480] [General]
I just added some code to makefile.mingw so it autogenerates a Windows Internet Shortcut file that links to the exact git revision/tag on github used to build exult when making a dist (if there is a tag it will use that otherwise the hash, if there is neither it just links to the repository). This is primarily for gpl reasons and intended for inclusion in the Windows Installer so there is always a link back to the source. One minor annoyance that I don't care enough to fix, snapshots get built before the tree is tagged so they will get a URL with a git hash to the revision instead of a slightly nicer looking url with the named tag

Making this a discussion incase others might want to give their input on things

I'm not sure if this sort of thing might be a good idea for other distributions even a source distro might want to have  link back to the exact revision it was made from.

Also wondering if it might be a nice idea to pass the git hash/tag to the compiler so Exult itself can printout the URL to the source. Nice to have for GPL but also nice to know the exact git revision a build belongs too

**Comment by DominusExult** [comment]
Thanks this is a really good idea. I am doing the same for the macOS snapshots (*) but as I do these on each code commit I don't use the snapshot release tag, but the git revision at the time of building.
![image](https://github.com/exult/exult/assets/3313896/6a88ea2b-d52d-4a33-a1f8-d20d926deacb)

* the macOS snapshots are not being done via a github workflow as to get a downward compatible build is a bit more elaborate for macOS and would really take a long time to build. Especially for Exult Studio. Factoring in two arches for that, is just too much and prone to breakage that has nothing to do with Exult itself.

**Comment by wench** [comment]
I pushed changes to add this to the Exult executable, Exult should now output the github source url on at startup as well as the git revision hash, tag, and remote branch. It seems to work for everything except ios because I have no idea how the ios build process works and if I made changes, I have no way to actually test it properly. This of course only works properly if Exult is built from within a git repository ideally with the normal exult github repository as the remote. It does not work if building from a source tarball so defaults to a url to this repository but no revision/tag

**Comment by DominusExult** [comment]
Great! and it works on macOS.

I'll have to look into iOS. As the iOS bild doesn't use the makefile stuff at all, I need to add a pre-build step that generates the gitinfo.h file and add that as a "maybe" header file (if that is even needed the way it is added in version.cc). Probably similar to what you needed to do for the Visual Studio build.

Can you add this information to the version screen (`v` in game)? 
And maybe just the (shortened?) revision on the cheat screen after the version number. On mobile systems you don't have that many keys but through the shortcutbar you can access the cheat menu.

**Comment by wench** [comment]
OK yeah I'll add it to the ingame displays

**Comment by wench** [comment]
Ok done. Not entirely happy with the way the Version scroll gump turned out. There is just such a limited line length.

**Comment by DominusExult** [comment]
done for iOS as well! (just not knowing where to add it for Android)

**Comment by wench** [comment]
Android uses the makefiles and should already include it. the logs for ci-andoind indicate it is generating gitinfo.h

**Reply by DominusExult** [reply]
Thanks! 🤦‍♂️ not that I haven't struggled the last few days with the Android build system and know it uses the makefiles🤦‍♂️ momentary mind blackout 🤪
